src/utils/plottableElement.ts

```diff
@@ -0,0 +1,55 @@
+export interface IPlottableElement<D extends Element> {
+    appendChild(newChild: Node): Node;
+    getBoundingClientRect(): ClientRect;
+    setAttribute(name: string, value: string): this;
+    left(position: number): this;
+    top(position: number): this;
+}
+
+export abstract class PlottableElement<D extends Element> implements IPlottableElement<D> {
```

Comment:
I like keeping classes and their related interface in the same file .. not sure if we want to enforce a pattern of separating them.